integrations/spark/src/main/java/marquez/spark/agent/lifecycle/SparkSQLExecutionContext.java

```diff
@@ -73,24 +77,43 @@ public void start(SparkListenerJobStart jobStart) {
       log.info("No execution info {}", queryExecution);
       return;
     }
+    UnknownEntryFacetListener unknownEntryFacetListener = new UnknownEntryFacetListener();
+    PartialFunction<LogicalPlan, List<Dataset>> outputFunc = PlanUtils.merge(outputDatasetSupplier);
     List<Dataset> outputDatasets =
-        PlanUtils.applyFirst(outputDatasetSupplier, queryExecution.optimizedPlan());
+        PlanTraversal.<LogicalPlan, List<Dataset>>builder()
+            .processor(outputFunc)
+            .processedElementListener(unknownEntryFacetListener)
+            .build()
+            .apply(queryExecution.optimizedPlan());
+
+    PartialFunction<LogicalPlan, List<Dataset>> inputFunc = PlanUtils.merge(inputDatasetSupplier);
     List<Dataset> inputDatasets =
         JavaConversions.seqAsJavaList(
-                queryExecution.optimizedPlan().collect(PlanUtils.merge(inputDatasetSupplier)))
+                queryExecution
+                    .optimizedPlan()
+                    .collect(
+                        PlanTraversal.<LogicalPlan, List<Dataset>>builder()
+                            .processor(inputFunc)
+                            .processedElementListener(unknownEntryFacetListener)
+                            .build()))
             .stream()
             .flatMap(List::stream)
             .collect(Collectors.toList());
+
+    UnknownEntryFacet unknownFacet =
+        unknownEntryFacetListener.build(queryExecution.optimizedPlan());
     LineageEvent event =
         LineageEvent.builder()
             .inputs(inputDatasets)
             .outputs(outputDatasets)
             .run(
                 buildRun(
                     buildRunFacets(
-                        buildLogicalPlanFacet(queryExecution.optimizedPlan()),
-                        null,
-                        buildParentFacet())))
+                        buildParentFacet(),
+                        AdditionalFacetWrapper.compose(
+                            "spark.logicalPlan",
```

Comment:
I think we should move to follow [custom facet naming](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.md#custom-facet-naming)

Reply:
so, we should go with 'SparkLogicalPlanRunFacet'? 
P.S. it was there already, I've just move it up

Reply:
And "spark_logicalPlan" key style, unless we already need to keep current names. @collado-mike do you know something about it?

Reply:
The existing spark facets were in place before we discussed any naming conventions. I think they ought to be migrated to match the expected conventions.